src/validations/html/htmlInspector.js

```diff
@@ -38,8 +39,37 @@ const errorMap = {
       },
     };
   },
+  'invalid-list-children': (error) => {
+    const tag = error.context.tagName.toLowerCase();
+
+    return {
+      reason: 'invalid-list-children',
+      payload: {
+        tag,
+        children: 'li',
+      },
+    };
+  },
 };
 
+function invalidListChildrenValidator(listener, reporter) {
+  listener.on('element', (elementName, domElement) => {
+    if (domElement.childNodes.length &&
+        (elementName === 'ul' || elementName === 'ol')) {
+      // Iterate over direct descendents of lists
+      for (let node = domElement.firstChild; node; node = node.nextSibling) {
```

Comment:
Is there an advantage to this style of iteration rather than the more straightforward `for (const node of domElement.childNodes)`?